plugins/crypto/mbedtls/ua_pki_mbedtls.c

```diff
@@ -471,6 +475,35 @@ certificateVerification_verify(void *verificationContext,
         } else {
             retval = UA_STATUSCODE_BADSECURITYCHECKSFAILED;
         }
+
+#ifdef UA_ENABLE_CERT_REJECTED_DIR
+        if(ci->rejectedListFolder.length !=0) {
+            char rejectedFileName[256] = {0};
+            UA_ByteString thumbprint;
+            UA_ByteString_allocBuffer(&thumbprint, UA_SHA1_LENGTH);
+            if(mbedtls_thumbprint_sha1(certificate, &thumbprint) == UA_STATUSCODE_GOOD) {
+                static const char hex2char[] = "0123456789ABCDEF";
+                for (size_t pos = 0, namePos = 0; pos < thumbprint.length; pos++) {
+                    rejectedFileName[namePos++] = hex2char[(thumbprint.data[pos] & 0xf0) >> 4];
+                    rejectedFileName[namePos++] = hex2char[thumbprint.data[pos] & 0x0f];
+                }
+                strcat(rejectedFileName, ".der");
+            } else {
+                UA_UInt32 dt = (UA_UInt32)UA_DateTime_now();
+                sprintf(rejectedFileName, "cert%d.der", dt);
```

Comment:
If an unsigned integer is printed to the file name, %u should be used instead of %d.